### PR TITLE
Added Radix Sort [Python]

### LINKED ISSUE #398: Radix Sort [Python]
#### This issue is about [Radix Sort](www.geeksforgeeks.org/radix-sort/).

- [x] Issue Title is in form `Algo/DS Name [Language]`.
- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [x] This Algo/DS is worth adding to this repository.
- [x] Assign this issue to me (I want to work on this).
	- [x] I ensure that I am not already assigned to 2 or more issues.
- [x] I will send my PR only after I'm assigned to this issue by any maintainer.
- [x] I understand that I will be unassigned from this issue if I fail to send a PR for this within a week.

### PR DESCRIPTION
**Pull request by Monal5031**
Fixes #398

#### By submitting this pull request I confirm I've read and complied with the below declarations.

- [x] I have read the [Contribution guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [coding guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#cs) for this project.
- [x] My code follows the [skeleton code structure](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#sample).
- [x] This pull request has a descriptive title. For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [x] This pull request will be closed if I fail to update it even once in a continuous time span of 7 days.

### CODE REVIEW

**Comment by aashutoshrathi** [comment]
Fix Travis Issues, Most of them are PEP8 and space-tab consistency related.

File: radix_sort/radix_sort.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+def main():
+	"""
+	Driver function to test radix sort.
+	"""
+	test = [170, 45, 75, 90, 802, 24, 2, 66]
+	print('Sorted array:', radix_sort(test))
+
+
+if __name__ == '__main__':
+    main()
```
Comment (aashutoshrathi) [inline radix_sort/radix_sort.py]: Add a newline at the end.

File: radix_sort/radix_sort.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        # split arr between lists
+        for i in arr:
+            tmp = i // digit
+            buckets[tmp % radix].append(i)
+            if max_length and tmp > 0:
+                max_length = False
+
+        # empty lists into arr array
+        a = 0
+        for b in range( radix ):
```
Comment (aashutoshrathi) [inline radix_sort/radix_sort.py]: Why spaces around `radix` ?

File: radix_sort/radix_sort.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
+# Following is the implementation of Radix Sort.
+
+
+def radix_sort(arr, radix=10):
+    """
+    :param arr: Iterable to sort.
```
Comment (aashutoshrathi) [inline radix_sort/radix_sort.py]: Better if you make it : `Iterable of elements to be sorted.`

File: radix_sort/radix_sort.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    Space complexity: O(n + k)
+    where, k is the range of input.
+    """
+    max_length = False
+    tmp , digit = -1, 1
+
+    while not max_length:
+        max_length = True
+        # declare and initialize buckets
+        buckets = [[] for _ in range( radix )]
```
Comment (aashutoshrathi) [inline radix_sort/radix_sort.py]: Why spaces around `radix` ?

**Comment by Monal5031** [comment]
@aashutoshrathi Updated the PR please take a look.

File: radix_sort/radix_sort.py [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    where, k is the range of input.
+    """
+    max_length = False
+    tmp, digit = -1, 1
+
+    while not max_length:
+        max_length = True
+        # declare and initialize buckets
+        buckets = [[] for _ in range(radix)]
+
```
Comment (Anshuman-Verma) [inline radix_sort/radix_sort.py]: Are we allowed to have blank lines in functions ?

Comment (Monal5031) [inline radix_sort/radix_sort.py]: I separated declarations and main algo with this line, making it a little more readable. I think its good to separate major things via blank lines. your views?

Comment (Anshuman-Verma) [inline radix_sort/radix_sort.py]: It does make code more readable, :+1:

File: radix_sort/radix_sort.py [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+def radix_sort(arr, radix=10):
+    """
+    :param arr: Iterable of elements to sort.
+    :param radix: Base of input numbers
+    :return: Sorted list of input.
+    Time complexity: O(d * (n + b))
+    where, n is the size of input list.
+           b is base of representation
+           d is number of digits in largest number
```
Comment (aashutoshrathi) [inline radix_sort/radix_sort.py]: `d is log(number) / log(base) `

Comment (Monal5031) [inline radix_sort/radix_sort.py]: And what does that give? @aashutoshrathi I mean you know math more than me I guess 😜

Comment (aashutoshrathi) [inline radix_sort/radix_sort.py]: Basically wanted to say please mention that `number of digits in that base`.

**Pull request merged**